### PR TITLE
Add default value support to Field interface

### LINKED ISSUE #17: Add default value support to Field interface
### Summary
Add support for default values in the Field interface to allow fields to automatically initialize with predefined values when they become visible, eliminating the need for field-specific logic in form handling code.

### Current Behavior
Fields that require default values (e.g., radio buttons that should always have a selection) need custom handling in widgetWizard.tsx.
This approach:

- Requires adding specific checks for each field that needs a default
- Separates default values from field definitions
- Makes the checkDependency function grow with each new field

### Expected Behavior

- Fields with defaultValue automatically get that value when they become visible
- Fields with defaultValue get that value when loading configs that don't have the field
- Fields with existing values preserve those values
- Fields without defaultValue behave as before

### Test Cases

- Create new recommendation widget → rank: "affinity" in config
- Load existing widget with rank: "popular" → preserves "popular"
- Load existing widget without rank field → adds rank: "affinity"

### PR DESCRIPTION
**Pull request by prosdev**
Closes #17 

Implement generic field defaults. Add testing infrastructure with some field logic tests

This pull request introduces a unified testing setup using Vitest workspaces, modernizes test dependencies, and improves the experience-editor package with comprehensive tests and enhanced default value handling for form fields. The most important changes are grouped below.

**Testing Infrastructure Modernization**

* Added a comprehensive `TESTING.md` guide covering test setup, commands, best practices, and troubleshooting for Vitest workspace-based testing.
* Switched the main test scripts in `package.json` from Nx to Vitest workspace commands (`test`, `test:watch`, `test:ui`, `tdd`, `cover`), while retaining Nx alternatives for compatibility.
* Updated test dependencies: upgraded `@testing-library/react`, added `@testing-library/jest-dom` and `jsdom` for improved React and DOM testing support. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L54-R66) [[2]](diffhunk://#diff-3b5f4420a327eb20faa618cb041803a8508e4afd11ef262da921a90c08a0d683R1)

**Experience Editor Improvements**

* Added a thorough test suite for the `RadioGroupInput` component, covering rendering, selection, interaction, and edge cases, ensuring robust UI behavior.
* Enhanced the form field logic in `widgetWizard.tsx` to automatically apply a field's `defaultValue` if no value is present, using new utility functions for clarity and maintainability. [[1]](diffhunk://#diff-154dacb4d59f235eb9ef4f353211ec52e59dab481c14f4410f2dd753cffda6efR96) [[2]](diffhunk://#diff-154dacb4d59f235eb9ef4f353211ec52e59dab481c14f4410f2dd753cffda6efR340-R344) [[3]](diffhunk://#diff-154dacb4d59f235eb9ef4f353211ec52e59dab481c14f4410f2dd753cffda6efL567-R579)
* Extended the `Field` interface and relevant field definitions to support a `defaultValue` property, and set a default for `contentRank` to "affinity". [[1]](diffhunk://#diff-5d442ab35f157526b6eab03d6a48e9eb1571847f50c9d8be425ea152462883d4R100) [[2]](diffhunk://#diff-c6faf6746b0055480de3a2f3e63940c04cc8ce1a5f5e2a1a13bab52d02f02604R97)

### CODE REVIEW

File: packages/experience-editor/src/components/widgetWizard.tsx [hunk 1]
```diff
@@ -564,19 +570,13 @@ const WidgetWizard: React.FC<WidgetWizardProps> = ({
...
-              }
-              return prevFormValues;
-            });
+          // Set default value for fields that define one
+          const dependentField = findFieldById(fields, id);
+          if (shouldApplyDefaultValue(dependentField, formValues[id])) {
+            setFormValues((prevFormValues) => ({
+              ...prevFormValues,
+              [id]: dependentField.defaultValue,
+            }));
```
Comment (ashyablok) [inline packages/experience-editor/src/components/widgetWizard.tsx]: 🎉

**Review by markhayden — APPROVED** [review]
approved assuming @ashyablok 's comment about the dupe functions gets resolved

File: packages/experience-editor/src/utility/fieldLogic.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  });
+
+  return Array.from(allFieldIds);
+}
+
+/**
+ * Determines which fields should be kept visible and which should be hidden
+ * based on the current value of a field with dependencies
+ */
+export function calculateFieldVisibility(
```
[the remark this answers is not in the document]
Comment (prosdev) [reply packages/experience-editor/src/utility/fieldLogic.ts]: Great point @ashyablok. I'll remove this -- as it's currently not needed/unused, along with several others (only used in tests)
